### PR TITLE
[release-1.6] issue-567: Support to add ResoureManagerTags to GCP Filestore resources

### PR DESCRIPTION
**Pull request by bharath-b-rh**
This is a manual cherry-pick of https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/pull/630

```release-note
Driver now supports adding resource tags to the filestore Instance resources created. Users can provide a comma separated list of tags using the new argument --resource-tags.
```

leiyiz: note to reviewers: this is not a pure backport anymore.

### CODE REVIEW

**Comment by bharath-b-rh** [comment]
@pwschuurman / @leiyiz Could you please help me with the PR review. This is a cherry-pick of the changes merged in master.

File: pkg/cloud_provider/user_tags.go [hunk 1]
```diff
@@ -0,0 +1,536 @@
...
+
+	// httpClient is currently used for mocking client for tests.
+	httpClient *http.Client
+	// httpEndpoint is the endpoint used for tests and can be used
+	// with httpClient only.
+	httpEndpoint string
+	// withoutAuthentication is currently used only for tests. It can
+	// also be utilized for accessing public resources.
+	withoutAuthentication bool
+}
```
Comment (bharath-b-rh) [inline pkg/cloud_provider/user_tags.go:108]: This change is not present in master and https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/pull/905 is created for adding the same. This is a new option added specifically for UTs to remove creds requirement.

Comment (bharath-b-rh) [reply pkg/cloud_provider/user_tags.go:108]: @pwschuurman^^ @leiyiz^^

**Comment by leiyiz** [comment]
let's keep the cherry pick PRs clean and don't change them. if there's bug fix needed let's just do separate cherry-picks

**Comment by bharath-b-rh** [comment]
> let's keep the cherry pick PRs clean and don't change them. if there's bug fix needed let's just do separate cherry-picks.
> 
> if too much change happened and we can't backport cleanly it might be good to consider cutting a new release branch

@leiyiz Thank you reviewing the changes!

I have reverted the additional changes. The cherry picks are clean and now contains only the changes from the source commit.

**Comment by bharath-b-rh** [comment]
@leiyiz ^^

**Review by leiyiz — COMMENTED** [review]
/approve

**Pull request merged**